crates/viewer/re_viewer_context/src/drag_and_drop.rs

```diff
@@ -0,0 +1,184 @@
+
+                        ui.horizontal(|ui| {
+                            ui.spacing_mut().item_spacing.x = 2.0;
+
+                            ui.small_icon(icon, Some(text_color));
+                            ui.label(egui::RichText::new(payload.to_string()).color(text_color));
+                        });
+                    })
+                    .response
+                })
+                .response;
+
+            let delta = pointer_pos - response.rect.right_bottom();
+            ui.ctx()
+                .transform_layer_shapes(layer_id, emath::TSTransform::from_translation(delta));
+        }
+    }
+}
+
+fn drag_pill_frame(error_state: bool) -> egui::Frame {
+    let hue = if error_state { Hue::Red } else { Hue::Blue };
+
+    egui::Frame {
+        fill: re_ui::design_tokens().color(ColorToken::new(hue, S325)),
+        stroke: egui::Stroke::new(
+            1.0,
+            re_ui::design_tokens().color(ColorToken::new(hue, S375)),
+        ),
+        rounding: (2.0).into(),
+        inner_margin: egui::Margin {
+            left: 6.0,
+            right: 9.0,
+            top: 5.0,
+            bottom: 4.0,
+        },
+        ..Default::default()
+    }
+}
+
+fn items_to_string<'a>(items: impl Iterator<Item = &'a Item>) -> String {
+    let mut container_cnt = 0u32;
+    let mut view_cnt = 0u32;
+    let mut app_cnt = 0u32;
+    let mut data_source_cnt = 0u32;
+    let mut store_cnt = 0u32;
+    let mut entity_cnt = 0u32;
+    let mut instance_cnt = 0u32;
+    let mut component_cnt = 0u32;
+
+    for item in items {
+        match item {
+            Item::Container(_) => container_cnt += 1,
+            Item::View(_) => view_cnt += 1,
+            Item::AppId(_) => app_cnt += 1,
+            Item::DataSource(_) => data_source_cnt += 1,
+            Item::StoreId(_) => store_cnt += 1,
+            Item::InstancePath(instance_path) | Item::DataResult(_, instance_path) => {
+                if instance_path.is_all() {
+                    entity_cnt += 1;
+                } else {
+                    instance_cnt += 1;
+                }
+            }
+            Item::ComponentPath(_) => component_cnt += 1,
+        }
+    }
+
+    let counts = [
+        &container_cnt,
+        &view_cnt,
+        &app_cnt,
+        &data_source_cnt,
+        &store_cnt,
+        &entity_cnt,
+        &instance_cnt,
+        &component_cnt,
+    ];
+
+    let names = [
+        ("container", "containers"),
+        ("view", "views"),
+        ("app", "apps"),
+        ("data source", "data sources"),
+        ("store", "stores"),
+        ("entity", "entities"),
+        ("instance", "instances"),
+        ("component", "components"),
+    ];
+
+    counts
+        .iter()
+        .zip(names.iter())
+        .filter_map(|(&&cnt, &name)| {
+            if cnt > 0 {
+                Some(format!(
+                    "{} {}",
+                    cnt,
```

Comment:
unlikely needed, but just for good practice
```suggestion
                    re_format::format_uint(cnt),
```